### DISCUSSION TITLE
Protocol Deviation Type KRI

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#542] [General]
This thread is to discuss the details of the Protocol Deviation Type KRI(s). General idea is to assess protocol deviation rate for 4 pre-specified categories: 
- Eligibility
- Medical
- Procedural
- Safety

@gwu05 Do you have thoughts about statistical methods here? Would be very straightforward to just add 4 new workflows that filter the data before calling `PD_Assess`, but definitely open to a more robust combined approach if needed.

**Comment by gwu05** [comment]
As discussed, the Protocol Deviation rate assessment, stratified by key protocol deviation categories versus calculating the % of each type of protocol deviations out of total number of deviations, should usually both be able to flag an aberrant site, but there may be some differences as they are technically testing different things (e.g. similar to testing overall all means are the same versus pairwise test of means and where the distribution may be different when the rate is similar/not-flagged and vice versa).

 - One example of this is if for a given site, the rate of Safety deviations occurred is say 10 events every month and is equal to the average rate for the overall/other sites (also 10 events every month). But the number of total deviations is only 10 for that site is only 10, so 100% of the deviations are Safety deviations, while the average overall/other sites is only 10% of deviations are Safety related. Then we would be still want the site to be flagged.

- Likewise, the alternative situation is if the % of deviations is roughly the same at 10% of Safety deviations out of all the protocol deviations for the site and for the other sites/overall. Then if the time in which the 10 events occurred is short like 10 events in 1 week, while the overall rate is 10 events every month, then the rate would pick-up on an issue. Although in this scenario, usually the overall protocol deviation rate could catch this.

One can imagine different scenarios like the above, where I think the key comes down to what question is being asked:
- Do we care that the rate of a specific protocol deviation type is different at a site?
- Or do we care that there are unusual large or small %s of protocol deviations type(s) at a site?

One last different is stratifying will result in X number of extra KRIs for each protocol deviation type, while doing a chi-squared or fisher's exact on the % of each protocol deviation type is one 'KRI' / test.

**Reply by gwu05** [reply]
Here's a nice reference discussing this as well on what denominator to use and the impact: https://www.appliedclinicaltrialsonline.com/view/defining-quality-tolerance-limits-and-key-risk-indicators-that-detect-risks-in-a-timely-manner-reflections-from-early-adopters-on-emerging-best-practices-part-3

**Comment by gwu05** [comment]
Another note to make is that the type of protocol deviations vary across studies.

For example, one study had:
![image](https://user-images.githubusercontent.com/64222539/175430832-0c996194-9638-4c4f-8851-342b3db781f3.png)

Another had:
![image](https://user-images.githubusercontent.com/64222539/175430849-538e3e75-72f2-4d58-b59e-9f054f793b90.png)

Another had:
![image](https://user-images.githubusercontent.com/64222539/175430896-e5889102-e0bf-4aef-840f-e5a837b7d0b7.png)

Another had:
![image](https://user-images.githubusercontent.com/64222539/175430915-a6f56e82-2578-4c3b-8e3e-b58c222de8b2.png)


It is potentially possible to map these to a common set of deviation types, but to do it automatically may require extra effort (otherwise this falls in the bucket of things that need to be manually mapped as input for study-specific info). The general we could consider may be: 

- Eligibility
- Informed Consent
- Treatment
- Medication
- Assessments / Procedures
- Missed visits and data
- Other

**Comment by jwildfire** [comment]
Plan add a workflow to run this stratified in v1.1 and then discuss. We can look into a more robust consilidated KRI as part of v1.2 based on user feedback.
